pkg/kubelet/kubeletconfig/util/codec/codec.go

```diff
@@ -78,10 +81,35 @@ func NewYAMLEncoder(groupName string) (runtime.Encoder, error) {
 
 // DecodeKubeletConfiguration decodes a serialized KubeletConfiguration to the internal type
 func DecodeKubeletConfiguration(kubeletCodecs *serializer.CodecFactory, data []byte) (*kubeletconfig.KubeletConfiguration, error) {
+	var (
+		obj runtime.Object
+		gvk *schema.GroupVersionKind
+	)
+
 	// the UniversalDecoder runs defaulting and returns the internal type by default
 	obj, gvk, err := kubeletCodecs.UniversalDecoder().Decode(data, nil, nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode, error: %v", err)
+		// For now allow lenient decoding without strict serializer. This
+		// path is to be dropped when v1beta1 is deprecated and removed.
+		if runtime.IsStrictDecodingError(err) {
+			strictErr := err
+			lenientScheme := runtime.NewScheme()
+			if err = kubeletconfig.AddToScheme(lenientScheme); err != nil {
+				return nil, fmt.Errorf("failed to add kubelet config API to lenient scheme: %v", err)
+			}
+			if err := kubeletconfigv1beta1.AddToScheme(lenientScheme); err != nil {
+				return nil, fmt.Errorf("failed to add kubelet config v1beta1 API to lenient scheme: %v", err)
+			}
+			lenientCodecs := serializer.NewCodecFactory(lenientScheme)
+			obj, gvk, err = lenientCodecs.UniversalDecoder().Decode(data, nil, nil)
+			if err != nil {
+				return nil, fmt.Errorf("failed lenient decoding: %v", err)
+			}
+			klog.Warningf("using lenient decoding as strict decoding failed: %v", strictErr)
```

Comment:
We should probably log this before attempting the decode?

Reply:
Fixed